### PR TITLE
renovate: Filter updates of PHP non-dev deps

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

For our non-dev PHP deps, we can't update if the upstream package has dropped support for the lowest PHP version we still support. Adjust renovate's config to filter those out.

Also, update the rule that sets `rangeStrategy: widen` for PHP dev deps to do it generally instead of depending on a list of specific deps. Too bad there's no "only widen if necessary for version support" option.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

None

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Merge this in a forked repo, add any necessary secrets, and run it. Or just look at https://github.com/anomiex/jetpack/actions/runs/7022848363/job/19108043107 where I did that.

### CODE REVIEW

File: .github/renovate-config.js [hunk 1]
```diff
@@ -81,5 +81,14 @@ module.exports = {
 			} )(),
 			enabled: false,
 		},
+		// PHP non-dev deps need to work with the oldest PHP versions we support.
+		{
+			matchDatasources: [ 'packagist' ],
```
Comment (jeherve) [inline .github/renovate-config.js:86]: When we match for `packagist`, wouldn't that also match all our own packages, and not just third-party packages? Would that be an issue?

Comment (anomiex) [reply .github/renovate-config.js:86]: Our own packages are all excluded just above.

OTOH, I just thought that we might want to restrict this to just _plugin_ deps, as that's the only place where we bundle stuff. So I might see if I can rework it later along those lines.

**Review by jeherve — APPROVED** [review]

**Comment by anomiex** [comment]
Darn, I forgot to turn off auto-merge...

File: .github/renovate-config.js (hunk 1 above)
Comment (anomiex) [reply .github/renovate-config.js:86]: Now that I think about it even more, it's probably better to leave this as-is. To make the change I was thinking we'd have to somehow ensure that the lower-version-compatible versions were what the build bundles in with the plugins, and then we'd have to make sure that all the CI tests run with those lower versions too. Better to just depend only on the lower versions in the first place, even in the packages.

**Pull request merged**